src/main/java/org/enginehub/worldgourd/bukkit/WorldGourdPlugin.java

```diff
@@ -109,12 +111,22 @@ public void onPlayerInteract(PlayerInteractEvent event) {
     @EventHandler(ignoreCancelled = true)
     public void onItemConsumption(PlayerItemConsumeEvent event) {
         Material item = event.getItem().getType();
-        if (item == Material.PUMPKIN_PIE) {
+        if (isGourd(item)) {
             sendMessage(event.getPlayer());
             event.setCancelled(true);
         }
     }
 
+    @EventHandler(ignoreCancelled = true)
+    public void onFrameRemoval(EntityDamageByEntityEvent event) {
+        if (event.getEntity() instanceof ItemFrame && isGourd(((ItemFrame) event.getEntity()).getItem().getType())) {
```

Comment:
Does any of this have the potential to be null?

Reply:
shouldn't, all of it has NotNull